+neuron/PlotShape.m

```diff
@@ -11,67 +11,32 @@
                 s = secs{i};
 
-                % Get all 3D point information.
-                pt3d = s.get_pt3d();
-
-                % Get total section length and arc points.
-                dx = double(s.length);
-                arc3d = pt3d(4, :);
-
-                % Iterate over segments.
-                segments = s.segments();
-                for j=1:s.nseg
-                    % Get start, pt3ds and end point for each segment.
-                    seg = segments{j};
-                    [x_lo, x_hi] = seg.get_bounds();
-                    x_lo = x_lo * dx;
-                    x_hi = x_hi * dx;
-                    seg_arc_arr = arc3d(arc3d(:) > x_lo & arc3d(:) < x_hi);
-                    seg_arc_arr = [x_lo seg_arc_arr x_hi];
-
-                    % Interpolate x,y,z,d for each point in seg_arc_arr.
-                    seg_x3d_arr = interp1_arr(pt3d(4, :), pt3d(1, :), seg_arc_arr);
-                    seg_y3d_arr = interp1_arr(pt3d(4, :), pt3d(2, :), seg_arc_arr);
-                    seg_z3d_arr = interp1_arr(pt3d(4, :), pt3d(3, :), seg_arc_arr);
-                    seg_d3d_arr = interp1_arr(pt3d(4, :), pt3d(5, :), seg_arc_arr);
-
-                    % Find value of selected quantity for this segment.
-                    var = spi.varname();
-                    seg_value = s.ref(var, seg.x).get();
-                    seg_value_rel = seg_value - spi.low();
-                    seg_value_rel = seg_value_rel  / (spi.high() - spi.low());
-                    seg_value_rel = min(max(seg_value_rel, 0), 1);
-                    
-                    % Plot between pt3ds, one pair at a time.
-                    for k=1:numel(seg_arc_arr)-1
-                        x = [seg_x3d_arr(k) seg_x3d_arr(k+1)];
-                        y = [seg_y3d_arr(k) seg_y3d_arr(k+1)];
-                        z = [seg_z3d_arr(k) seg_z3d_arr(k+1)];
-                        plot_struct = struct;
-                        plot_struct.x = x;
-                        plot_struct.y = y;
-                        plot_struct.z = z;
-                        plot_struct.line_width = (seg_d3d_arr(k) + seg_d3d_arr(k+1))/2;
-                        plot_struct.color = [seg_value_rel, 0, 1-seg_value_rel];
-                        data{end+1} = plot_struct;
-                    end
-                end
+                section_plot_data = double(clib.neuron.get_section_plot_data(s.sec, spi));
+                section_plot_data = transpose(reshape(section_plot_data, 9, []));
+
+                x = [section_plot_data(:, 1) section_plot_data(:, 2)];
+                y = [section_plot_data(:, 3) section_plot_data(:, 4)];
+                z = [section_plot_data(:, 5) section_plot_data(:, 6)];
+                d = (section_plot_data(:, 7) + section_plot_data(:, 8)) / 2;
+                v = section_plot_data(:, 9);
+                plot_table = table(x, y, z, d, [v, zeros(size(v)), 1-v], 'VariableNames', {'x', 'y', 'z', 'line_width', 'color'}); 
```

Comment:
Concatenating tables now.
Using the plotting of a table in a loop is really slow (e.g. plot3(tbl, 'x', 'y', 'z') in the loop), adding about 10 seconds.
And a single plot3 can still not be done since it doesn't allow for line width and / or color specification.